lua/utils/security.lua

```
@@ -0,0 +1,278 @@
+    vim.notify("Failed to write to file: " .. (write_err or "unknown error"), vim.log.levels.ERROR)
+    return false
+  end
+  
+  return true
+end
+
+-- Check if a path is within allowed directories
+-- @param path Path to check
+-- @return boolean True if path is allowed
+function M.is_path_allowed(path)
+  local norm_path = vim.fn.fnamemodify(path, ":p")
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'vim'

(W113)